### PR TITLE
Make stop checking for equations constantly

### LINKED ISSUE #5590: Organelle tooltips shouldn't have ChemicalEquation `AutoRefreshProcess` on
as the numbers only change when switching patches, so this unnecessarily eats up performance in the editor

### PR DESCRIPTION
**Pull request by Patryk26g**
**Brief Description of What This PR Does**

Redid my  pr because I messed up the previous one.
Adds ChemicalEquations to the global group and call ls the UpdateEquation function on patch change. Hopefully this solution is acceptable

**Related Issues**

Closes #5590

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by hhyyrylainen** [comment]
You can force push over existing PRs. So there is no need to create a new PR to even replace every single commit in the PR. If you can't push normally use force pushing to force overwrite commits in a PR.

File: src/microbe_stage/gui/PatchDetailsPanel.cs [hunk 1]
```diff
@@ -595,6 +595,7 @@ private void MoveToPatchClicked()
             return;
         }
 
+        GetTree().CallGroup("ChemicalEquations", "UpdateEquation");
```
Comment (hhyyrylainen) [inline src/microbe_stage/gui/PatchDetailsPanel.cs:598]: I'm sorry to say but I don't like this solution. We don't use Godot groups for anything so adding it for just this one thing seems not good.

My preferred solution is still to trigger a GUI tooltip update in the editor in the patch changed callback. It needs a few levels of indirection but it shouldn't take that long to write all the proxy methods that get the notice to update where it is needed.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/microbe_stage/gui/PatchDetailsPanel.cs (hunk 1 above)
Comment (Patryk26g) [reply src/microbe_stage/gui/PatchDetailsPanel.cs:598]: Do we have some similar solution somewhere in the code? Would be easier to get inspired by it :D

Comment (hhyyrylainen) [reply src/microbe_stage/gui/PatchDetailsPanel.cs:598]: I can't immediately think of an exact match but `UpdateMPCost` in `CellEditorComponent.GUI.cs` has a loop that updates a property on all tooltips. So basically that but instead just calling a method on each tooltip that would then in turn call a method on all chemical equations that the tooltip contains (this is likely the only slightly tricky part as the tooltips need to be updated to know about the chemical equations inside them).

**Comment by Patryk26g** [comment]
Well just turning off the automatic process update on the organelleSelectionTooltip was enough.

I was also thinking about the processes that are displayed during the microbe game. When we display them they start updating constantly even when the processes list is hidden so we probably would need an issue for that

**Comment by hhyyrylainen** [comment]
> Well just turning off the automatic process update on the organelle selection tooltip was enough.

What updates the tooltips then now to make sure they don't show outdated information?

> I was also thinking about the processes that are displayed during the microbe game. When we display them they start updating constantly even when the processes list is hidden so we probably would need an issue for that

Easiest thing would be if the process panel just deleted its contents when it is hidden but I don't consider this much of a problem as after all there's a limited set of possible processes the player cell might have so that caps the performance impact and many players never open the process panel so they wouldn't be impacted.

**Comment by Patryk26g** [comment]
> What updates the tooltips then now to make sure they don't show outdated information?

`WriteOrganelleProcessList` is called in `UpdateOrganelleEfficiencies` which is called in 'CalculateOrganelleEffectivenessInCurrentPatch' which in turn is called whenever the patch or light level is changed

**Review by hhyyrylainen — APPROVED** [review]
I did quite a bit of testing and I think I can confirm that the process tooltip rebuild method is always called when necessary so the changes here to disable the auto-refresh on the tooltips should be exactly right and this PR should now be good to go as solving the issue.

**Pull request merged**